### PR TITLE
Item menu option to favorite and unfavorite items

### LINKED ISSUE #28061: Mark favorite items in inventory
**Is your feature request related to a problem? Please describe.**  
It is common to carry a certain set of items in inventory at all times. Might be a lighter to start fires, a flashlight, a lock-pick, aspirin, syringe of adrenaline, etc. It varies from player to player and from situation to situation. It is also common to load your backpack, pockets and any other place that can hold a milliliter with the random, not immediately useful, temporary loot that is to be dropped later at your truck or base.

**Describe the solution you'd like**  
Add an option to mark an item as favorite. Favorite items appear a different color in inventory. Later that can be expanded to "drop all but favorite" functionality.

**Describe alternatives you've considered**  
Perhaps some integration with auto pickup but this is not for pickup (might be for drop-off later). It is purely visual.

**Additional context**  
I am keeping the scope intentionally small: only a color change. I want to try to code this myself. If requested additional functionality ("drop all but favorite") can be added later.

### PR DESCRIPTION
**Pull request by IronRex**
Adds item menu option to `f` favorite and unfavorite items. Favorite items will appear a different color in inventory screens.

#### Summary
```SUMMARY: Interface "Adds ability to mark favorite items in inventory"```

#### Purpose of change
Closes #28061

#### Describe the solution
Added `f` `favorite` action to item actions in inventory screens. Becomes `f` `unfavorite` action for items that are already marked as favorite.

#### Additional context
I realize that `unfavorite` is not exactly English but it is short and understandable. LMK if you have a better idea for `favorite`/`unfavorite` menu text.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -2105,7 +2105,15 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
         addentry( 'p', pgettext( "action", "part reload" ), u.rate_action_reload( oThisItem ) );
         addentry( 'm', pgettext( "action", "mend" ), u.rate_action_mend( oThisItem ) );
         addentry( 'D', pgettext( "action", "disassemble" ), u.rate_action_disassemble( oThisItem ) );
+
+        if (oThisItem.is_favorite) {
```
Comment (Night-Pryanik) [inline src/game.cpp]:
```suggestion
        if( oThisItem.is_favorite ) {
```

**Comment by Xhuis** [comment]
I was just thinking about something like this. Fantastic addition.

**Comment by dissociativity** [comment]
is it possible to move favorited items to their own category at the top of the inventory?

**Comment by IronRex** [comment]
IMO that would go against existing categorization.

It is definitely out of scope of this change or changes I might try to make after this change (such as drop all but favorite).

You can open another issue for that of course.

**Comment by dyingsoundsfun** [comment]
I do apologize for this mistake as well. Want to have lame list of weak or
borderline pathetic excuses? Then again it was honestly an oopsies...

**Comment by IronRex** [comment]
Can someone clue me in to what Travis and Gorgon want?

**Comment by OrenAudeles** [comment]
Gorgon looks mad because of a JSON linting error (unrelated to PR afaict) in `data/json/items/armor/boots.json`, and Travis is mad for the same reason. To see yourself:
Gorgon 
```
Click Details
On Left side panel open "Console Log"
Scroll to bottom
```
Travis-CI
```
Click Details
Click on a RED colored entry in build jobs
Once JOB LOG loads, scroll to bottom to see why it failed
```

**Comment by Night-Pryanik** [comment]
Could you please provide a screenshot of how this system work?

**Comment by IronRex** [comment]
This is a normal inventory view
image
Item menu has an option to favorite an item
image
Favorite item shows with an asterisk and in brighter color
image
Favorite items have option to unfavorite
image
Items otherwise colored (unread books, ammo, etc) retain their natural color. Only default color is changed for favorite items. All favorite items, regardless of color, show asterisk after their normal name string.

File: src/game.cpp [hunk 2]
```diff
@@ -2208,6 +2216,9 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
                 case 'D':
                     u.disassemble( pos );
                     break;
+                case 'f':
+                    u.favorite( pos );
```
Comment (kevingranade) [inline src/game.cpp]: As oThisItem is still in scope, can't we just toggle is_favorite field directly instead of going through the player?

**Pull request merged**